### PR TITLE
PREAPPS-4934: Move notifications handling from api-js to x-web

### PR DESCRIPTION
**Pull request by yashrajpchavda**
- moved the notifications handler to zm-x-web, updated necessary code in api-js for the same pupose
- Improvements in the notifications code to be done in a separate ticket

### CODE REVIEW

File: src/batch-client/index.ts [hunk 1]
```diff
@@ -1477,12 +1480,16 @@ export class ZimbraBatchClient {
...
 
-			if (notifications && this.notificationHandler) {
+			if (notifications && this.notificationsEmitter) {
 				// as notification handling happens in synchronous way, if the notifications count is really higher (for bulk operations)
 				// the UI would freeze because of the JavaScript execution time. Hence, delayed the notification handling to give some time
 				// for the render to happen in the JavaScript event loop
 				setTimeout(() => {
-					this.notificationHandler && this.notificationHandler(notifications);
+					this.notificationsEmitter &&
+						this.notificationsEmitter.emit(
```
Comment (silentsakky) [inline src/batch-client/index.ts:1489]: curious, if we are using event emitter then do we need setTimeout wrapping it? as I feel event emitter works on event loop only

File: src/schema/schema.ts [hunk 2]
```diff
@@ -41,7 +41,6 @@ import { ZimbraSchemaOptions } from './types';
 
 import { ZimbraBatchClient } from '../batch-client';
 import { coerceBooleanToString } from '../utils/coerce-boolean';
-import { ZimbraNotifications } from './notifications';
```
Comment (silentsakky) [inline src/schema/schema.ts:44]: should we delete this file as well?

File: src/batch-client/index.ts (hunk 1 above)
Comment (yashrajpchavda) [reply src/batch-client/index.ts:1489]: Yes now the setTimeout would be irrelevant, I will remove it...

File: src/schema/schema.ts (hunk 2 above)
Comment (yashrajpchavda) [reply src/schema/schema.ts:44]: Yep that was the last step to do, I will just remove it...

Comment (yashrajpchavda) [reply src/schema/schema.ts:44]: Done - deleted

File: src/batch-client/index.ts (hunk 1 above)
Comment (yashrajpchavda) [reply src/batch-client/index.ts]: Removed

Comment (silentsakky) [reply src/batch-client/index.ts]: awesome

**Pull request merged**